snapcraft/internal/deltas/_deltas.py

```diff
@@ -0,0 +1,214 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2016 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+
+import logging
+import os
+import shutil
+import subprocess
+import time
+
+from progressbar import ProgressBar
+
+from snapcraft import file_utils
+from snapcraft.internal.deltas.errors import (
+    DeltaFormatError,
+    DeltaFormatOptionError,
+    DeltaGenerationError,
+    DeltaToolError,
+)
+
+
+logger = logging.getLogger(__name__)
+
+
+delta_formats_options = [
+    'xdelta'
+]
+
+
+class BaseDeltasGenerator:
+    """Class for delta generation
+
+    This class is responsible for the snap delta file generation
+    """
+
+    delta_format = None
+    delta_file_extname = 'delta'
+    delta_tool_path = None
+
+    def __init__(self, source_path, target_path):
+        self.source_path = source_path
+        self.target_path = target_path
+        self.pre_check()
+
+    def pre_check(self):
```

Comment:
Since `pre_check` is only used in `__init__` maybe just get rid of it and make it part of `__init__`